src/Core/Grid/Definition/Factory/CustomerAddressGridDefinitionFactory.php

```diff
@@ -54,7 +54,19 @@ final class CustomerAddressGridDefinitionFactory extends AbstractGridDefinitionF
     public function __construct(HookDispatcherInterface $hookDispatcher, ?Request $currentRequest)
     {
         parent::__construct($hookDispatcher);
-        $this->backUrl = $currentRequest ? $currentRequest->getUri() : '';
+
+        // Remove useless query parameters to avoid some strange behaviors with delete workflow
+        // This way, behavior of "customer address grid" is consistent with `AddressGridDefinitionFactory`
+        // (should be solved using proper front-end and API architecture)
+
+        $query = $currentRequest->query;
```

Comment:
As `$currentRequest` can be `null`, you should check it before trying to access `query`